### PR TITLE
Add reverse function for array

### LINKED ISSUE #5636: Please add a way to reverse sort an array
Right now presto only supports array_sort which sorts array in ascending order, it would be really helpful for users to have ability to sort array in descending order. It can be an argument within array_sort or array_reverse function call.

### PR DESCRIPTION
**Pull request by geraint0923**
Implement ARRAY_REVERSE function

Closes #5636

### CODE REVIEW

File: presto-docs/src/main/sphinx/functions/array.rst [hunk 1]
```diff
@@ -53,6 +53,10 @@ Array Functions
 
     Sorts and returns the array ``x``. The elements of ``x`` must be orderable.
 
+.. function:: array_reverse(x) -> array
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/array.rst]: Put this in alphabetic order

File: presto-docs/src/main/sphinx/functions/array.rst [hunk 2]
```diff
@@ -53,6 +53,10 @@ Array Functions
...
+
+    Returns a array which has the reversed order of the given array.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/functions/array.rst]: "an array"

File: presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java [hunk 3]
```diff
@@ -501,6 +501,18 @@ public void testSort()
     }
 
     @Test
+    public void testReverse()
+            throws Exception
+    {
+        assertFunction("ARRAY_REVERSE(ARRAY[1,2,3,4])", new ArrayType(INTEGER), ImmutableList.of(4, 3, 2, 1));
+        assertFunction("ARRAY_REVERSE(ARRAY_SORT(ARRAY[2, 3, 4, 1]))", new ArrayType(INTEGER), ImmutableList.of(4, 3, 2, 1));
+        assertFunction("ARRAY_REVERSE(ARRAY[2, BIGINT '3', 4, 1])", new ArrayType(BIGINT), ImmutableList.of(1L, 4L, 3L, 2L));
+        assertFunction("ARRAY_REVERSE(ARRAY['a','b','c','d'])", new ArrayType(createVarcharType(1)), ImmutableList.of("d", "c", "b", "a"));
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java]: Add space after comma to improve readability

File: presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java [hunk 4]
```diff
@@ -501,6 +501,18 @@ public void testSort()
...
+        assertFunction("ARRAY_REVERSE(ARRAY[TRUE, FALSE])", new ArrayType(BOOLEAN), ImmutableList.of(false, true));
+        assertFunction("ARRAY_REVERSE(ARRAY[1.1,2.2,3.3,4.4])", new ArrayType(DOUBLE), ImmutableList.of(4.4, 3.3, 2.2, 1.1));
```
Comment (electrum) [inline presto-main/src/test/java/com/facebook/presto/type/TestArrayOperators.java]: Same

**Comment by geraint0923** [comment]
Addressed the comments.

**Comment by cberner** [comment]
@electrum & @martint do you want to call this `array_reverse`, `reverse`, `reversed` or something else?

**Comment by cberner** [comment]
Looks good except for deciding on the name

**Comment by martint** [comment]
`reverse` seems good. We already have it for `varchar`, and since the semantics are similar it's a good fit.

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayReverseFunction.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        pageBuilder = new PageBuilder(ImmutableList.of(elementType));
+    }
+
+    @TypeParameter("E")
+    @SqlType("array(E)")
+    public Block reverse(
+            @TypeParameter("E") Type type,
+            @SqlType("array(E)") Block block)
+    {
+        int arrayLength = block.getPositionCount();
```
Comment (fmeiser) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayReverseFunction.java]: You could use a shortcut if the array has less than 2 elements.

**Comment by geraint0923** [comment]
Rename to `reverse` and address the comment.

**Pull request merged**